app/src/pages/contract/[chain]/[address].tsx

```diff
@@ -164,6 +139,164 @@ export default function Address() {
     [addressInfos, addressInfosSearch]
   )
 
+  const connectWallet = async function () {
+    // Check if the ethereum object exists on the window object
+    if (typeof window.ethereum !== 'undefined') {
+      try {
+        await window.ethereum.enable()
+
+        const provider = new ethers.providers.Web3Provider(window.ethereum)
+
+        setSigner(provider.getSigner())
+      } catch (error) {
+        console.error('User rejected request', error)
+      }
+    } else {
+      console.error('Metamask not found')
+    }
+  }
```

Comment:
This shouldn't be dependent on metamask. In the future we should migrate it to the wagmi native functions, similar to [Attestoooooor.tsx](https://github.com/ethereum-optimism/optimism-starter/blob/24b6e60ad500af5ad20939d0cc644ce9991a5824/src/components/Attestoooooor.tsx#L2) example. But it enough for the demo

Reply:
will change it later :)